Index: kwalletopener.cpp
===================================================================
--- kwalletopener.cpp	(Revision 1147487)
+++ kwalletopener.cpp	(Arbeitskopie)
@@ -30,7 +30,6 @@
 #include <KToolInvocation>
 
 #include <QtCore/QTextStream>
-#include <QtCrypto/QtCrypto>
 
 #include <unistd.h>
 
Index: opener.h
===================================================================
--- opener.h	(Revision 1147487)
+++ opener.h	(Arbeitskopie)
@@ -24,7 +24,7 @@
 #include <QtCore/QString>
 #include <QtCore/QFile>
 #include <QtCore/QTextStream>
-#include <QtCrypto/QtCrypto>
+#include <qca2/QtCrypto/QtCrypto>
 #include <QtDBus/QDBusServiceWatcher>
 #include <KDebug>
 #include <KLocalizedString>
