diff -ru kdepim-runtime-21.12.3/CMakeLists.txt kdepim-runtime-21.12.3-patched/CMakeLists.txt
--- kdepim-runtime-21.12.3/CMakeLists.txt	2022-02-07 07:10:36.000000000 +0100
+++ kdepim-runtime-21.12.3-patched/CMakeLists.txt	2022-04-08 12:42:23.131258436 +0200
@@ -105,7 +105,7 @@
 if(Qt5TextToSpeech_FOUND)
 set(HAVE_TEXT_TO_SPEECH_SUPPORT TRUE)
 endif()
-find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Network Widgets Test XmlPatterns DBus WebEngineWidgets NetworkAuth)
+find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Network Widgets Test XmlPatterns Concurrent DBus WebEngineWidgets NetworkAuth)
 
 
 # KF5 package
diff -ru kdepim-runtime-21.12.3/resources/etesync/CMakeLists.txt kdepim-runtime-21.12.3-patched/resources/etesync/CMakeLists.txt
--- kdepim-runtime-21.12.3/resources/etesync/CMakeLists.txt	2022-02-07 07:10:36.000000000 +0100
+++ kdepim-runtime-21.12.3-patched/resources/etesync/CMakeLists.txt	2022-04-08 12:42:46.388579134 +0200
@@ -51,6 +51,7 @@
 endif()
 target_link_libraries(akonadi_etesync_resource
     ${ETEBASE_LIBRARIES}
+    Qt::Concurrent
     Qt::DBus
     KF5::AkonadiAgentBase
     KF5::ConfigCore
