--- kdepim-runtime-23.04.1.orig/CMakeLists.txt  2023-05-09 05:37:36.000000000 +0200
+++ kdepim-runtime-23.04.1/CMakeLists.txt       2023-05-25 10:27:12.507850634 +0200
@@ -113,7 +113,7 @@
 if(Qt${QT_MAJOR_VERSION}TextToSpeech_FOUND)
 set(HAVE_TEXT_TO_SPEECH_SUPPORT TRUE)
 endif()
-find_package(Qt${QT_MAJOR_VERSION} ${QT_REQUIRED_VERSION} CONFIG REQUIRED Network Widgets Test DBus NetworkAuth)
+find_package(Qt${QT_MAJOR_VERSION} ${QT_REQUIRED_VERSION} CONFIG REQUIRED Network Widgets Concurrent Test DBus NetworkAuth)
 if (QT_MAJOR_VERSION STREQUAL "5" AND BUILD_TESTING)
     find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED XmlPatterns)
 endif()


--- kdepim-runtime-23.04.1.orig/resources/etesync/CMakeLists.txt        2023-05-25 10:19:37.308137453 +0200
+++ kdepim-runtime-23.04.1/resources/etesync/CMakeLists.txt     2023-05-25 10:21:55.128754972 +0200
@@ -52,6 +52,7 @@
 target_link_libraries(akonadi_etesync_resource
     ${ETEBASE_LIBRARIES}
     Qt::DBus
+    Qt::Concurrent
     KPim${KF_MAJOR_VERSION}::AkonadiAgentBase
     KF${KF_MAJOR_VERSION}::ConfigCore
     KF${KF_MAJOR_VERSION}::Contacts

