diff -ru kdepim-runtime-21.08.3/CMakeLists.txt kdepim-runtime-21.08.3-patched/CMakeLists.txt
--- a/CMakeLists.txt	2021-11-01 03:20:22.000000000 +0100
+++ b/CMakeLists.txt	2021-12-01 09:18:03.647328298 +0100
@@ -101,7 +101,7 @@
                    PURPOSE "Needed for ews resource.")
 option(KDEPIM_RUN_AKONADI_TEST "Enable autotest based on Akonadi." TRUE)
 # QT5 package
-find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED TextToSpeech Network Widgets Test XmlPatterns DBus WebEngineWidgets NetworkAuth)
+find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED TextToSpeech Network Widgets Test XmlPatterns Concurrent DBus WebEngineWidgets NetworkAuth)
 
 
 # KF5 package
diff -ru kdepim-runtime-21.08.3/resources/etesync/CMakeLists.txt kdepim-runtime-21.08.3-patched/resources/etesync/CMakeLists.txt
--- a/resources/etesync/CMakeLists.txt	2021-11-01 03:20:22.000000000 +0100
+++ b/resources/etesync/CMakeLists.txt	2021-12-01 09:18:58.164399969 +0100
@@ -51,6 +51,7 @@
 endif()
 target_link_libraries(akonadi_etesync_resource
     ${ETEBASE_LIBRARIES}
+    Qt::Concurrent
     Qt::DBus
     KF5::AkonadiAgentBase
     KF5::ConfigCore
