diff -ru kdepim-runtime-22.04.3/CMakeLists.txt kdepim-runtime-22.04.3.patched/CMakeLists.txt
--- kdepim-runtime-22.04.3/CMakeLists.txt	2022-07-13 08:12:04.060954505 +0200
+++ kdepim-runtime-22.04.3.patched/CMakeLists.txt	2022-07-13 08:15:58.362636518 +0200
@@ -105,7 +105,7 @@
 if(Qt5TextToSpeech_FOUND)
 set(HAVE_TEXT_TO_SPEECH_SUPPORT TRUE)
 endif()
-find_package(Qt${QT_MAJOR_VERSION} ${QT_REQUIRED_VERSION} CONFIG REQUIRED Network Widgets Test DBus NetworkAuth)
+find_package(Qt${QT_MAJOR_VERSION} ${QT_REQUIRED_VERSION} CONFIG REQUIRED Network Widgets Concurrent DBus NetworkAuth)
 if (QT_MAJOR_VERSION STREQUAL "5")
     find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED XmlPatterns)
 endif()
diff -ru kdepim-runtime-22.04.3/resources/etesync/CMakeLists.txt kdepim-runtime-22.04.3.patched/resources/etesync/CMakeLists.txt
--- kdepim-runtime-22.04.3/resources/etesync/CMakeLists.txt	2022-07-13 08:12:03.974286032 +0200
+++ kdepim-runtime-22.04.3.patched/resources/etesync/CMakeLists.txt	2022-07-13 08:15:08.561573336 +0200
@@ -52,6 +52,7 @@
 target_link_libraries(akonadi_etesync_resource
     ${ETEBASE_LIBRARIES}
     Qt${QT_MAJOR_VERSION}::DBus
+    Qt${QT_MAJOR_VERSION}::Concurrent
     KF5::AkonadiAgentBase
     KF5::ConfigCore
     KF5::Contacts
